cirq/google/serializable_device_test.py

```diff
@@ -264,4 +265,52 @@ def test_mixing_types():
     gate.valid_targets.extend(['subset', 'sym'])
 
     with pytest.raises(NotImplementedError):
-        _ = cg.SerializableDevice.from_proto(proto=spec, gate_set=_JUST_MEAS)
+        _ = cg.SerializableDevice.from_proto(proto=spec, gate_sets=[_JUST_MEAS])
+
+
+def test_multiple_gatesets():
+    halfPiGateSet = cirq.google.serializable_gate_set.SerializableGateSet(
+        gate_set_name='half_pi_gateset',
+        serializers=cgc.SINGLE_QUBIT_HALF_PI_SERIALIZERS,
+        deserializers=cgc.SINGLE_QUBIT_HALF_PI_DESERIALIZERS,
+    )
+    allAnglesGateSet = cirq.google.serializable_gate_set.SerializableGateSet(
+        gate_set_name='half_pi_gateset',
+        serializers=cgc.SINGLE_QUBIT_SERIALIZERS,
+        deserializers=cgc.SINGLE_QUBIT_DESERIALIZERS,
+    )
+    durations_dict = {'xy_pi': 20_000, 'xy_half_pi': 20_000, 'xy': 20_000}
+    spec = cirq.google.known_devices.create_device_proto_from_diagram(
+        "aa\naa", [allAnglesGateSet, halfPiGateSet], durations_dict)
+    dev = cg.SerializableDevice.from_proto(
+        proto=spec, gate_sets=[allAnglesGateSet, halfPiGateSet])
+    dev.validate_operation(cirq.X(cirq.GridQubit(0, 0)))
+    dev.validate_operation(cirq.X(cirq.GridQubit(1, 0)))
```

Comment:
Should validate an operation with arbitrary angle as well.

Reply:
Done